components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -51,14 +54,19 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
                 directory = Environment.CurrentDirectory;
             }
 
-            _availableJobs = new Queue<JobProcessingInstructionModel>();
+            _availableJobs = new ConcurrentQueue<JobProcessingInstructionModel>();
             _assignedJobs = new ConcurrentDictionary<string, JobProcessingInstructionModel>();
 
+            _lock = new SemaphoreSlim(1, 1);
+
+            RefreshJobFromFile();
+
             var file = Path.GetFileName(_legacyCliModel.PublishedNodesFile);
             _fileSystemWatcher = new FileSystemWatcher(directory, file);
             _fileSystemWatcher.Changed += _fileSystemWatcher_Changed;
+            _fileSystemWatcher.Created += _fileSystemWatcher_Changed;
+            _fileSystemWatcher.Renamed += _fileSystemWatcher_Changed;
             _fileSystemWatcher.EnableRaisingEvents = true;
```

Comment:
ok got it, this is the fix for the issue where filesystemwatcher died silently...